server/app/controllers/ti/TrustedIntermediaryController.java

```diff
@@ -80,26 +83,26 @@ public Result dashboard(
     if (trustedIntermediaryGroup.isEmpty()) {
       return notFound();
     }
+    SearchParameters searchParameters =
+        SearchParameters.builder().setSearch(search).setSearchDate(searchDate).build();
     ImmutableList<Account> managedAccounts =
-        trustedIntermediaryGroup.get().getManagedAccounts(search, searchDate);
+        tiService.retriveSearchResult(searchParameters, trustedIntermediaryGroup);
```

Comment:
Consider naming this method `getManagedAccounts` or `filterManagedAccounts`. `retrieveSearchResult` seems a bit general to know what's being returned.